dd-sdk-android/src/main/kotlin/com/datadog/android/rum/gestures/DatadogGesturesListener.kt

```diff
@@ -0,0 +1,149 @@
+package com.datadog.android.rum.gestures
+
+import android.view.GestureDetector
+import android.view.MotionEvent
+import android.view.View
+import android.view.ViewGroup
+import com.datadog.android.core.internal.utils.sdkLogger
+import com.datadog.android.tracing.Tracer
+import java.lang.ref.WeakReference
+import java.util.LinkedList
+import java.util.concurrent.TimeUnit
+
+internal class DatadogGesturesListener(
+    private val rumTracer: Tracer,
+    private val decorViewReference: WeakReference<View>
+) :
+    GestureDetector.OnGestureListener {
+
+    // region GesturesListener
+
+    override fun onShowPress(e: MotionEvent) {
+        // No Op
+    }
+
+    override fun onSingleTapUp(e: MotionEvent): Boolean {
+        decorViewReference.get()?.let {
+            // we just intercept but not steal the event
+            val spanBuilder = rumTracer
+                .buildSpan(UI_TAP_ACTION_EVENT)
+            val target =
+                findTarget(
+                    it,
+                    e.x,
+                    e.y
+                )
+            spanBuilder.withTag(TAG_TARGET_CLASS_NAME, target.javaClass.canonicalName)
+            // we will handle general exceptions here. We don't know what might go wrong apart from
+            // ResourceNotFound exception.
+            @Suppress("TooGenericExceptionCaught")
+            try {
+                spanBuilder.withTag(
+                    TAG_TARGET_RESOURCE_ID,
+                    target.resources.getResourceEntryName(target.id)
+                )
+            } catch (e: Exception) {
+                sdkLogger.e("$TAG: Could not found resource name for target:${target.id}", e)
```

Comment:
cant we default to `target.id` rather than doing nothing? 
`val target.tag: String = try {resources.getResourceEntryName(id) } catch(...) id.toString() `